infra/helper.py

```diff
@@ -160,6 +160,18 @@ def main():
                                              help='Pull base images.')
 
   args = parser.parse_args()
+
+  # We have different default values for `sanitizer` depending on the `engine`.
+  try:
+    if not args.sanitizer:
+      if args.engine == 'dataflow':
+        args.sanitizer = 'dataflow'
+      else:
+        args.sanitizer = 'address'
+  except:
```

Comment:
Why is this needed now?

Reply:
Because some commands do not have `sanitizer` argument and line 166 throws an exception.

Reply:
Which commands? Why is this a new development?

Reply:
e.g. `build_image` doesn't require `--sanitizer`:

```
$ python infra/helper.py build_image --no-pull zstd
Traceback (most recent call last):
  File "infra/helper.py", line 852, in <module>
    sys.exit(main())
  File "infra/helper.py", line 166, in main
    if not args.sanitizer:
AttributeError: 'Namespace' object has no attribute 'sanitizer'
```

It wasn't an issue as we previously wouldn't attempt to access `args.sanitizer` unless we knew it's specified (e.g. when `build_fuzzers` command is used). Now I have to access it to assign the default value dynamically. Unfortunately `argparse` doesn't seem to have any safe method like `.get(key)` to check if the value is present or not :|

Reply:
What I mean is that this script makes a lot of effort to exit with a nonzero return code if any command it runs exits with nonzero. But the exit value of this script [is ignored](https://github.com/google/oss-fuzz/blob/4d3073292ae43d99b1bd34b58e45021cb83760ae/infra/base-images/base-runner/test_all#L59) in favor of hackily looking for output. Can you leave a TODO to fix this please?

Reply:
Btw, you can probably use `hasattr` instead of the current approach of catching an exception.

Reply:
Oooh, nice! Thanks! Done.